pyfar/dsp/dsp.py

```diff
@@ -223,3 +119,77 @@ def nextpow2(x):
         Exponent of next higher power of 2.
     """
     return np.ceil(np.log2(x))
+
+
+def spectrogram(signal, dB=True, log_prefix=20, log_reference=1,
+                window='hann', window_length=1024, window_overlap_fct=0.5):
+    """Compute the magnitude spectrum versus time.
+
+    This is a wrapper for scipy.signal.spectogram with two differences. First,
+    the returned times refer to the start of the FFT blocks, i.e., the first
+    time is always 0 whereas it is window_length/2 in scipy. Second, the
+    returned spectrogram is normalized accroding to `signal.signal_type` and
+    `signal.fft_norm`.
+
+    Parameters
+    ----------
+    signal : Signal
+        pyfar Signal object.
+    db : Boolean
+        Falg to plot the logarithmic magnitude specturm. The default is True.
+    log_prefix : integer, float
+        Prefix for calculating the logarithmic time data. The default is 20.
+    log_reference : integer
+        Reference for calculating the logarithmic time data. The default is 1.
+    window : str
+        Specifies the window (See scipy.signal.get_window). The default is
+        'hann'.
+    window_length : integer
+        Specifies the window length in samples. The default ist 1024.
+    window_overlap_fct : double
+        Ratio of points to overlap between fft segments [0...1]. The default is
+        0.5
+
+    Returns
+    -------
+    frequencies : numpy array
+        Frequencies in Hz at which the magnitude spectrum was computed
+    times : numpy array
+        Times in seconds at which the magnitude spectrum was computed
+    spectrogram : numpy array
+    """
+
+    # check input
+    if not isinstance(signal, Signal):
+        raise TypeError('Input data has to be of type: Signal.')
+
+    if window_length > signal.n_samples:
+        raise ValueError("window_length exceeds signal length")
+
+    # get spectrogram from scipy.signal
+    window_overlap = int(window_length * window_overlap_fct)
+    window = sgn.get_window(window, window_length)
+
+    frequencies, times, spectrogram = sgn.spectrogram(
+            x=signal.time.squeeze(), fs=signal.sampling_rate, window=window,
+            noverlap=window_overlap, mode='magnitude', scaling='spectrum')
+
+    # remove normalization from scipy.signal.spectrogram
+    spectrogram /= np.sqrt(1 / window.sum()**2)
+
+    # apply normalization from signal
+    spectrogram = fft.normalization(
+        spectrogram, window_length, signal.sampling_rate,
+        signal.fft_norm, window=window)
+
+    # get in dB
+    if dB:
+        eps = np.finfo(float).tiny
+        spectrogram = log_prefix*np.log10(
+            np.abs(spectrogram) / log_reference + eps)
```

Comment:
I think I would not implement the dB option here, but in the respective plot function.